### ISSUE TITLE
Main thread is throwing random_state error.

### ISSUE THREAD

**Issue by Akhilj786** [hyperopt/hyperopt#790]
Error log:
 [2021-03-16 19:31:52,377] [49/MainThread] [ERROR] [hyperopt.fmin] job exception: __init__() got an unexpected keyword argument 'random_state'
 
 Sample code:
 ```
 def optimize_lightfm(optimization_params, train):

    def objective(fn_params):
        num_epochs = fn_params.pop("num_epochs")
        model = lightfm.LightFM(**fn_params)
        model.fit(train, epochs=num_epochs)
        score = lfm_precision_at_k(model, train, k=params.K_FOR_PR_AT_K).mean()
        return {"loss": -score, "status": STATUS_OK}

    tpe_trials = Trials()
    tpe_best = fmin(
        fn=objective,
        space=optimization_params,
        algo=tpe.suggest,
        trials=tpe_trials,
        max_evals=params.MAX_EVAL,
        rstate=np.random.RandomState(params.RANDOM_SEED),
    )
    return tpe_best
```
We evaluate this function in concurrent future fashion to compute multiple models and pick the best model.
  
Version Info: 
OS: Ubuntu

Python version: 3.6
scikit-learn==0.24.1
hyperopt==0.2.5

Can somebody help to address this issue?

**Comment by Akhilj786** [comment]
On further debugging we identified we were running implicit on an older version i.e  0.3.7. 
Previously, I was using pip to install implicit== 0.4.4, switched to conda forge which broke this functionality.
https://anaconda.org/conda-forge/implicit/labels (see gcc7 vs main).
Switched to `main` on conda forge, the functionality started working again.
